src/main/java/edu/cornell/kfs/paymentworks/service/impl/PaymentWorksWebServiceImpl.java

```diff
@@ -68,27 +74,46 @@
 	private String directoryPath;
 	
 	protected ClientRequest buildClientRequest(String url) {
-        return buildClientRequest(url, null);
-    }
-	
+		ClientRequest.Builder builder = new ClientRequest.Builder();
+		builder.header(PaymentWorksConstants.AUTHORIZATION_HEADER_KEY, buildAuthorizationHeaderString());
+		builder.accept(MediaType.APPLICATION_XML);
+
+		return builder.build(buildURI(url), HttpMethod.GET);
+	}
+
 	protected ClientRequest buildClientRequest(String url, String jsonString) {
-        ClientRequest.Builder builder = new ClientRequest.Builder();
-        builder.accept(MediaType.APPLICATION_XML);
-        builder.header("Authorization", "Token " + paymentworksAuthorizationToken);
-        if (ObjectUtils.isNotNull(jsonString)) {
-        	//.put(Entity.entity(jsonString, MediaType.APPLICATION_JSON));
-        	builder.entity(jsonString, MediaType.APPLICATION_JSON);
-        }
-        
-        URI uri;
+		ClientRequest.Builder builder = new ClientRequest.Builder();
+		builder.header(PaymentWorksConstants.AUTHORIZATION_HEADER_KEY, buildAuthorizationHeaderString());
+		builder.accept(MediaType.APPLICATION_JSON);
+		builder.entity(jsonString, MediaType.APPLICATION_JSON);
+
+		return builder.build(buildURI(url), HttpMethod.PUT);
+	}
+
+	protected ClientRequest buildClientRequest(String url, MultiPart multiPart) {
+		ClientRequest.Builder builder = new ClientRequest.Builder();
+		builder.accept(MediaType.MULTIPART_FORM_DATA);
+		builder.header(PaymentWorksConstants.AUTHORIZATION_HEADER_KEY, buildAuthorizationHeaderString());
+		builder.entity(multiPart, MediaType.MULTIPART_FORM_DATA);
+
+		return builder.build(buildURI(url), HttpMethod.POST);
+	}
+
+	protected URI buildURI(String url) {
+		LOG.debug("buildURI The passed in URL is: " + url);
+		URI uri;
         try {
             uri = new URI(url);
         } catch (URISyntaxException e) {
+        	LOG.error("buildURI unable to build a URI object for the URL: '" + url +"' with the error: ", e);
```

Comment:
logging the error is good, but it would probably also be helpful to include a message when re-throwing the exception to provide additional context